frontend/packages/db-structure/src/schema/schema.ts

```diff
@@ -82,6 +82,37 @@ export type ForeignKeyConstraint = v.InferOutput<
   typeof foreignKeyConstraintSchema
 >
 
+/**
+ * Valid reference options for Google Cloud Spanner interleave constraints
+ *
+ * Spanner interleave constraints only support CASCADE and NO_ACTION for delete operations.
+ * Update operations are always NO_ACTION.
+ */
+const interleaveConstraintReferenceOptionSchema = v.picklist([
+  'CASCADE',
+  'NO_ACTION',
+])
+export type InterleaveConstraintReferenceOption = v.InferOutput<
+  typeof interleaveConstraintReferenceOptionSchema
+>
+
+/**
+ * Schema for Google Cloud Spanner interleave constraints
+ * @see {@link https://cloud.google.com/spanner/docs/schema-and-data-model#creating-interleaved-tables}
+ */
+export const interleaveConstraintSchema = v.object({
+  type: v.literal('INTERLEAVE'),
+  name: constraintNameSchema,
+  columnName: columnNameSchema,
+  targetTableName: tableNameSchema,
+  targetColumnName: columnNameSchema,
+  updateConstraint: v.literal('NO_ACTION'),
+  deleteConstraint: interleaveConstraintReferenceOptionSchema,
+})
+export type InterleaveConstraint = v.InferOutput<
+  typeof interleaveConstraintSchema
+>
```

Comment:
[memo]
I separated interleave schema from foreign key.

Reason
- There are some differences between foreign key and interleave schema like updateConstraint, type
  - we have to force interleave implementation to logic/type def for foreign key unintentionally
- There is actually little benefit to define common schema for foreign key and interleave